### DISCUSSION TITLE
请问文件名、目录名是否会被加密？

### DISCUSSION THREAD

**Discussion by gdtv** [juicedata/juicefs#4945] [Q&A] [answered]
在中文文档里 https://juicefs.com/docs/zh/cloud/encryption/ 有说明：

> JuiceFS 支持存储加密，即对数据进行加密后再上传到对象存储，对象存储中只有加密后的密文，可以有效地防止对象存储的访问权限泄露或者存储被入侵等情况下泄露数据隐私。
> 
> ......
> 
> 另外，JuiceFS 中的元数据目前不支持加密存储，请不要在元数据（文件名、目录名，或者扩展属性）中保存敏感信息（比如用户的 email 等隐私信息）。

但是在英文文档里 https://juicefs.com/docs/cloud/encryption/ ，却没有

> 另外，JuiceFS 中的元数据目前不支持加密存储，请不要在元数据（文件名、目录名，或者扩展属性）中保存敏感信息（比如用户的 email 等隐私信息）。

这一句话。

请问文件名、目录名是否会被加密？ 如果别人拿到我远程备份的文件，是否有可能获取我的文件名列表？

**Comment by jiefenghuang** [comment, marked as answer]
元数据内容没有加密, 社区版元数据引擎访问可以授权

**Reply by gdtv** [reply]
再请教下， 如果别人拿到我远程备份的文件，是否有可能获取我的原始文件名？

**Reply by jiefenghuang** [reply]
1. 文件数据是存储在对象存储, 文件是切片后上传的, 而且也支持加密, 这样如果有人访问到你的对象存储, 也访问不了原始文件内容
2. 文件元数据可以存储在redis/mysql等等, 这些访问你也可以加密(密码/密钥等). 如果有人可以访问你的redis元数据, 那是可以读取到信息, 比如文件名.
